admin/acceptance.py

```diff
@@ -950,6 +951,98 @@ def eliot_output(message):
     sys.stdout.write(format % message)
     sys.stdout.flush()
 
+def capture_upstart(reactor, host, output_file):
+    """
+    SSH into given machine and capture relevant logs, writing them to
+    output file.
+
+    :param reactor: The reactor.
+    :param bytes host: Machine to SSH into.
+    :param file output_file: File to write to.
+    """
+    formatter = TailFormatter(output_file, host)
+    ran = run_ssh(
+        reactor=reactor,
+        host=host,
+        username='root',
+        command=[
+            b'tail',
+            b'/var/log/flocker/flocker-control.log',
+            b'/var/log/flocker/flocker-dataset-agent.log',
+            b'/var/log/flocker/flocker-container-agent.log',
+            b'/var/log/flocker/flocker-docker-plugin.log',
+            b'/var/log/upstart/docker.log',
+        ],
+        handle_stdout=formatter.handle_output_line,
+    )
+    ran.addErrback(write_failure, logger=None)
+    # Deliver a final empty line to process the last message
+    ran.addCallback(lambda ignored: formatter.handle_output_line(b""))
+    return ran
+
+
+class TailFormatter(object):
+    """
+    Formatter for the output of the ``tail`` commands that will produce logs
+    with Eliot messages with the same format as the ones produced when
+    parsing journalctl output
+
+    :ivar file output_file: log file where we want to write our log
+    :ivar bytes _host: ip address or identifier of our host to be
+        added to the Eliot messages
+    :ivar bytes service: optional initial name of the service. This initial
```

Comment:
The regex thing is unnecessary and seems very fragile to me; you should just require this and have the caller pass the service name in since it knows it, by having a tail per-service (see above).